game/resources.h

```diff
@@ -210,11 +214,13 @@ class Resources final {
         }
       };
 
+
     Tempest::Device&                  dev;
     Tempest::SoundDevice              sound;
 
     std::recursive_mutex              sync;
     std::unique_ptr<Dx8::DirectMusic> dxMusic;
+    DmLoader*                         dmLoader;
```

Comment:
Since it's a C library, this is sort of required. The `DmLoader` is reference-counted under the hood so a `unique_ptr` would be semantically incorrect. Even if we ignored that, to make it a `unique_ptr` would require adding a custom de-allocation function which makes passing it around a pain. I'll do that though if you want me to :)

Reply:
Introducing a `.hpp` header, similarly to how it done in vulkan can be a good option.
In this hpp, you can place tiny C++ wrappers around C-api